registry/storage/driver/filesystem/driver.go

```diff
@@ -248,7 +248,7 @@ func (d *driver) Delete(ctx context.Context, subPath string) error {
 // URLFor returns a URL which may be used to retrieve the content stored at the given path.
 // May return an UnsupportedMethodErr in certain StorageDriver implementations.
 func (d *driver) URLFor(ctx context.Context, path string, options map[string]interface{}) (string, error) {
-	return "", storagedriver.ErrUnsupportedMethod
+	return "", new(storagedriver.ErrUnsupportedMethod)
```

Comment:
You can return a concrete value with the syntax `storagedriver.ErrUnsupportedMethod{}`. I don't recommend using pointer types for errors.